### ISSUE TITLE
Does not work - OSX

### ISSUE THREAD

**Issue by dan2735** [FPtje/gmosh#5]
http://pastebin.com/R7jwMJPG

Occurs on --help.

**Comment by Bo98** [comment]
Basically this:
```
$ otool -L bin/gmosh
bin/gmosh:
	/opt/local/Library/Frameworks/Python.framework/Versions/3.4/Python (compatibility version 3.4.0, current version 3.4.0)
	/opt/local/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 855.16.0)
```
should look like this (note the relative path):
```
$ otool -L bin/gmosh
bin/gmosh:
	@executable_path/Python (compatibility version 3.4.0, current version 3.4.0)
	@executable_path/libintl.8.dylib (compatibility version 10.0.0, current version 10.3.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 855.16.0)
```

This is usually corrected using Apple's `install_name_tool` but [cx_Freeze only seems to call that when running bdist_mac](https://bitbucket.org/anthony_tuininga/cx_freeze/src/1282b6b6ee637738210113dd88c3c198d475340f/cx_Freeze/macdist.py?at=default#cl-182) (creates a .app) and not a standard build (creates a standard command line executable). I opened an issue here: https://bitbucket.org/anthony_tuininga/cx_freeze/issue/129. It is possible to call `install_name_tool` in the makefile so it's not too much of a big deal.

However, there is a bug in cx_Freeze when the append_script_to_exe option is True. It creates a .zip with the required .pyc files and appends it to the executable but does not modify the Mach-O headers to cover the appended zip file. Because of this, `install_name_tool` fails to run with the error `the __LINKEDIT segment does not cover the end of the file (can't be processed)`. I opened another issue here: https://bitbucket.org/anthony_tuininga/cx_freeze/issue/118. The workaround for that is to set append_script_to_exe to False (for OS X at least). It creates a `library.zip` file but there's a several other files already being shipped anyway. Setting the option does require a small Python script poking cx_Freeze as the command line cxfreeze doesn't support setting append_script_to_exe.

It's also worth noting that `install_name_tool` requires Xcode developer tools to be installed (not the Xcode IDE itself). Simply running the command will prompt you about it. It's not exactly a small download  as it contains a lot (see 'files' section of [the xcode-select docs](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man1/xcode-select.1.html)), though it should be possible to create a Python script which reads and replaces library paths in the headers if avoiding the developer tools build dependency is preferrable.

I probably made this sound a lot more complex than it actually is. Basically setting append_script_to_exe to False and either manually calling `install_name_tool` or creating a Python script to read over the headers fixes the Python dependency.
